### ISSUE TITLE
Worth a try?

### ISSUE THREAD

**Issue by jbishop129** [michaelusner/esphome-components#1]
Hi @michaelusner I found this `external_component` work you put together, after I posted over in your other repo thread :) I was wondering if this is in a state that's worth trying out on my end? I'd be very happy to move my Pentair ESP32 + RS485 over to ESPHome for better lifecycle management, if it is able to pull the controller data just as well as the other full firmware project. Also happy to help test and contribute!

**Comment by jbishop129** [comment]
Well, I couldn't resist, so I tried :)
I can see it reading data off the Pentair UART bus, but it doesnt seem to quite have enough to parse and decode the strings for temp and device sensors; below is a sample output I receive on the console. 
When I crank up `debug:` logging on the `uart:` in ESPhome, I can see all the data barfing on the bus, but I unfortunately don't have enough background in this area to make heads or tails of what it all means!

```
[14:53:37][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:37][C][ota:096]: Over-The-Air Updates:
[14:53:37][C][ota:097]:   Address: pool-controller.local:3232
[14:53:37][C][ota:100]:   Using Password.
[14:53:37][C][ota:103]:   OTA version: 2.
[14:53:37][C][api:139]: API Server:
[14:53:37][C][api:140]:   Address: pool-controller.local:6053
[14:53:37][C][api:142]:   Using noise encryption: YES
[14:53:37][C][pentair_easytouch.component:100]: Pentair EasyTouch component
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.componentAir Temp:  'Air Temperature'
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.component  Device Class: 'temperature'
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.component  State Class: ''
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.component  Unit of Measurement: '°F'
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.component  Accuracy Decimals: 0
[14:53:37][C][pentair_easytouch.component:101]: pentair_easytouch.component  Icon: 'mdi:thermometer'
[14:53:37][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:39][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x07
length       = 0x00

[14:53:39][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x10
src          = 0x60
command      = 0x07
length       = 0x0f

[14:53:39][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:41][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:41][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:43][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:43][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:45][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:45][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x04
length       = 0x01

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x10
src          = 0x60
command      = 0x04
length       = 0x01

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x06
length       = 0x01

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x10
src          = 0x60
command      = 0x06
length       = 0x01

[14:53:47][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x01
length       = 0x04

[14:53:48][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x10
src          = 0x60
command      = 0x01
length       = 0x02

[14:53:48][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:49][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:49][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:51][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[14:53:53][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[14:53:53][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x07
length       = 0x00

[14:53:53][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x10
src          = 0x60
command      = 0x07
length       = 0x0f
```

**Comment by michaelusner** [comment]
Thanks for taking a look at this - it's encouraging to see you trying it out.  After replacing my equipment after the lightning strike (main board and pump), I added a ScreenLogic controller.  Now that I have control of the equipment on my phone, I have to admit that my motivation has diminished here.
Remodelling my office has thrown a wrench into things as well because I moved my desk into another room that doesn't have a drop for the RS485 connection to the controller.  
I'm just being lazy and would like to get it working though.  I need to drop the line in to the new office and find my breadboard to try to bring it up again.  I really appreciate your interest in this.

**Comment by jbishop129** [comment]
Very much appreciate your response, and all the development you've done on this so far!  I'm running this ESPhome in parallel to your existing and working one that is a full firmware; the original still works great, read-only.

On the ESPhome component, it does seem to be reading the bus, as evidenced by the serial console outputs, but doesnt seem to pick up the summary status on the bus. My understanding from all of your previous work is the Pentair controller broadcasts a global device status payload, it looks like every few seconds (I think that's the `dest = 0x0f` but not sure), but unfortunately it doesn't seem to be parsing that here. Or easily just as likely, I don't fully understand it :)

That said, I realize you've moved on to a different path with the ScreenLogic controller, so totally understandable if this project is not front of mind!

**Comment by jbishop129** [comment]
I've been digging into this as far as I can with my limited RS-485 skills, and wondered if you have some thoughts on this: When I enable `verbose` logging in ESPhome, I get lots of output such as the snippet below. It looks like we've got good parsed packets emitting from the `pentair_easytouch.component`, I'm just not sure why that doesnt translate into sensor outputs we can use to send to Home Assistant 🤔 


```
[15:54:59][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[15:55:00][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x90
src          = 0x10
command      = 0xd2
length       = 0x01

[15:55:01][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x33
dest         = 0x0f
src          = 0x10
command      = 0x02
length       = 0x1d

[15:55:02][V][pentair_easytouch.component:061]: leading_byte = 0xa5
type         = 0x00
dest         = 0x60
src          = 0x10
command      = 0x07
length       = 0x00
```

**Comment by michaelusner** [comment]
When I last had it working, all I had set up at the time was air and water
temperature. IIRC, I was having trouble with getting more components in
place because I didn't understand esp home's sdk well enough.

**Comment by jbishop129** [comment]
For what it’s worth, I opted to go a different direction here, and wanted to share my results. Not being a C++ dev or have any skills helpful in decoding UART RS485, I realized I wasn’t personally going to get anywhere on my own. Searching around, I came across the [nodejs-poolController](https://github.com/tagyoureit/nodejs-poolController) project, and learned that one of the hardware interface options is a “Elfin EW11 RS485‐WiFi” bridge. This wires straight into the RS485 bus, some quick configuration lands it on your IP Wi-Fi network, and is supported in nodejs-poolController. Awesome! I gritted my teeth while searching the Amazon for how many many hundreds of dollars this device might be, and was delighted to find it was $25 USD 😁 and would arrive at my door in 24 hours!

Setup was as simple as advertised, and I quickly had full control of my pool via dockerized poolController and poolController-Dashboard components. Last ToDo for myself is to plumb this back into Home Assistant via MQTT, a simple task.

I wanted to report this back in case anyone else is struggling with this as well. I do realize Michael you had some code contribution on that project, so I’m very much not telling you anything you dont already know!

For my case, this solution is fit for purpose and I’m satisfied with the result. There is certainly a path and a use case for the roll-your-own ESP32+RS48+ESPHome bridge, I just dont have the skills to implement it 🥲
